llvm/test/Transforms/PGOProfile/thinlto_indirect_call_promotion.ll

```diff
@@ -1,39 +0,0 @@
-; Do setup work for all below tests: generate bitcode and combined index
```

Comment:
Hmm, I think it is still preferred to have a test here as runtime tests are more sensitive to the environment and may be skipped by some contributors.

It means we will have to duplicate some file between here and the runtime test. Perhaps adding a comment can help connect the two tests.

Reply:
this is a valid point. Add the test back (in the new commit), use `rm -rf %t && split-file %s %t && cd %t` on IR and apply the `trap-on-exit` clean-up on the shell script.

Reply:
It looks like that we had 3 precanned profraw files and this patch is adding another one. Since the raw profile isn't stable, this is going to add some inconvenience whenever the version is bumped.
`thinlto_indirect_call_promotion.profraw` probably should be rewritten as `.proftext` and `llvm-profdata merge`
```
% fd -e profraw
Transforms/PGOProfile/Inputs/thinlto_indirect_call_promotion.profraw
tools/llvm-profdata/Inputs/basic.profraw
tools/llvm-profdata/Inputs/compressed.profraw
tools/llvm-profdata/Inputs/c-general.profraw
```

It seems that there is an earlier comment that proftext may not cover the use case well, so we also utilize a runtime test in `compiler-rt/test`.

Reply:
Without use of the raw profile, this test would not have caught the regression. If we think the new compiler-rt test is enough to catch this case in the future, then perhaps we don't need this LLVM test at all (there should already be some ICP tests). But if the compiler-rt test is not enough, because some people don't run those, then this should stay and use a raw profile as input.